internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go

```diff
@@ -184,7 +174,291 @@ func resourceDataProtectionBackupPolicyPostgreSQL() *pluginsdk.Resource {
+													// In view of possible support for `VaultStore` in the future, the `data_store_type` property is exposed for users to set in version 4.0.
+													// feedback from the service team: Theoretically all 3 values possible. But currently only logical combination is from VaultStore to ArchiveStore. So in target data store it can only be ArchiveStore. OperationalStore isn’t supported for this workload.
+													string(backuppolicies.DataStoreTypesArchiveStore),
+												}, false),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+	} else {
+		resource.Schema["retention_rule"].Elem.(*pluginsdk.Resource).Schema["life_cycle"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"data_store_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							// confirmed with the service team that the possible values do not include `OperationalStore`.
+							string(backuppolicies.DataStoreTypesVaultStore),
+							string(backuppolicies.DataStoreTypesArchiveStore),
+						}, false),
+					},
+
+					"duration": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validate.ISO8601Duration,
+					},
+
+					"target_copy": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"option_json": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringIsJSON,
+								},
+								"data_store_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										// since the following feedback from the service team, the current possible values only support `ArchiveStore`.
+										// However, in view of possible support for `VaultStore` in the future, the `data_store_type` property is exposed for users to set in version 4.0.
+										// feedback from the service team: Theoretically all 3 values possible. But currently only logical combination is from VaultStore to ArchiveStore. So in target data store it can only be ArchiveStore. OperationalStore isn’t supported for this workload.
+										string(backuppolicies.DataStoreTypesArchiveStore),
+									}, false),
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		resource.Schema["default_retention_rule"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"life_cycle": {
```

Comment:
would it make sense to just pull out lifecycle and instead just allow multiple default retention rule blocks?

Reply:
Hi @katbyte, thanks for your feedback. However, the schema of `default_retention_rule` in this PR is consistent with the schema of resources [azurerm_data_protection_backup_policy_kubernetes_cluster](https://github.com/hashicorp/terraform-provider-azurerm/blob/bca6d3d99e55ea7a2e22c644d6139903c9df5329/internal/services/dataprotection/azurerm_data_protection_backup_policy_kubernetes_cluster_resource.go#L104) and [azurerm_data_protection_backup_policy_postgresql_flexible_server ](https://github.com/hashicorp/terraform-provider-azurerm/blob/bca6d3d99e55ea7a2e22c644d6139903c9df5329/internal/services/dataprotection/data_protection_backup_policy_postgresql_flexible_server_resource.go#L95) in the same RP. Considering user habits and consistency, I assume that it might be better to keep it unchanged, what do you think? Thanks for your time and looking forward to your suggestions.

PS: I have updated the PR with 5.0 flags.